### DISCUSSION TITLE
Force enable tracking to bypass DNS detection issues

### DISCUSSION THREAD

**Discussion by deefdragon** [postalserver/postal#2529] [Feature suggestions]
## Describe the bug

I'm having some issues with cloudflare and setting up the DNS for tracking in postal. Whenever I get it correctly detected, it will eventually disable itself and stop tracking because of DNS failures.

Id like the ability to force tracking even if postal thinks the DNS is incorrect. 

It would also be nice to get some better errors on the "check DNS" button as even checking the logs I dont get anything, and debugging is mostly me just trying things until they work.

## To Reproduce

Set click.mydomain and track.postal.mydomain in cloudflare, both proxied.

## Expected behaviour

Better erroring on the check DNS to explain what is incorrect, and the ability to override if its functioning through all other testing.

## Environment details 

 - OS: [hosted on ubuntu in docker]
 - Browser [chrome]
 - Version [2.1.2]
 - Type [server]

**Comment by willpower232** [comment]
That is interesting, I didn't know that it checks the DNS status.

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/message_parser.rb#L11

It looks like the code records the error in the database, can you confirm what the error message recorded is?

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/app/models/track_domain.rb#L54-L70

I imagine the problem is that the code is explicitly looking for a CNAME and I believe cloudflare proxying probably emits A records. 

Is there any reason you're using proxying for this?

**Reply by deefdragon** [reply]
Trying to set up postal again and running into this same issue. To answer the question, I have everything in cloudflare. Ive tried disabling the proxying, to no success unfortunately. Regardless, I want to keep it behind proxy anyway. I see no reason that the click-tracking links should not be proxied and protected by cloudflare.

**Reply by willpower232** [reply]
After you disable proxying, can you run nslookup for that domain and see that cloudflare emits a CNAME? The code has changed a little in version 3 but still expects your click tracking domain to be a CNAME.

**Reply by deefdragon** [reply]
I did some more retrying and fighting with DNS caching, and _was_ able to get Cloudflare to output a cname. Unfortunately, Cloudflare then outputs an A record  of my IP address directly (as it is no longer proxied, this makes a little bit of sense), and the firewall I have set up only accepts traffic from Cloudflare's IP addresses. This results in the clicked link not functioning.

As mentioned previously, I would like to avoid disabling using Cloudflare's protections if possible.

**Comment by davaxi** [comment]
I Have same problem... :(
